docs(workflow): add consumer function security documentation (#1543)
vercel/workflow: docs/content/docs/deploying/world/vercel-world.mdx

- You don't need to add authentication or authorization logic to workflow handlers
- Unauthorized requests can never reach the step or workflow functions
- Only messages delivered through Vercel Queues can trigger execution
- Handlers receive only a message ID that must be retrieved from Vercel's backend, making it impossible to craft custom payloads

The trigger configuration also controls retry behavior. Both handlers use `retryAfterSeconds: 5` (compared to the default 60 seconds for regular queues) to keep workflow execution responsive.

| Option | Type | Default | Description |
| --- | --- | --- | --- |
| `type` | `string` | — | Trigger type. Workflows use `queue/v2beta` |
| `topic` | `string` | — | Topic name to consume. Supports wildcards (e.g., `__wkf_step_*`) |
| `consumer` | `string` | — | Consumer group name |
| `retryAfterSeconds` | `number` | 60s | Time before a failed message is retried |
| `initialDelaySeconds` | `number` | 0s | Delay before the consumer starts processing after deploy |

<Callout>

>   This configuration is managed entirely by the Workflow DevKit build step. You never need to write this yourself — it's documented here for transparency. For custom framework integrations, see [Framework Integrations — Security](/docs/how-it-works/framework-integrations#security).

Reviewer: ```suggestion
  This configuration is managed entirely by the Workflow DevKit build step. You should not need to write this yourself. If you are writing a custom integration, see [Framework Integrations — Security](/docs/how-it-works/framework-integrations#security) for more details..
```
